### PR TITLE
core: backport some ResultCode updates

### PR DESCRIPTION
**Pull request by GPUCode**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6645)

### CODE REVIEW

File: src/core/hle/result.h [hunk 1]
```diff
@@ -255,181 +254,138 @@ constexpr ResultCode UnimplementedFunction(ErrorModule module) {
...
+ * Code(), Succeeded() or Failed() prior to accessing the inner value.
  *
  * An example of how it could be used:
  * \code
  * ResultVal<int> Frobnicate(float strength) {
  *     if (strength < 0.f || strength > 1.0f) {
  *         // Can't frobnicate too weakly or too strongly
- *         return ResultCode(ErrorDescription::OutOfRange, ErrorModule::Common,
- *             ErrorSummary::InvalidArgument, ErrorLevel::Permanent);
+ *         return ResultCode{ErrorModule::Common, 1};
```
Comment (Steveice10) [inline src/core/hle/result.h:276]: Maybe I'm missing something, where did all of the other result code components go here?

File: src/core/hle/service/cecd/cecd.cpp [hunk 2]
```diff
@@ -1329,7 +1329,7 @@ void Module::CheckAndUpdateFile(const CecDataPathType path_type, const u32 ncch_
                 const u32 message_size = static_cast<u32>(message->GetSize());
                 std::vector<u8> buffer(message_size);
 
-                message->Read(0, message_size, buffer.data()).Unwrap();
+                void(message->Read(0, message_size, buffer.data()).Unwrap());
```
[the remark this answers is not in the document]
Comment (GPUCode) [reply src/core/hle/service/cecd/cecd.cpp:1332]: It ignores the result value even with the [[nodiscard]] attribute

File: src/core/hle/result.h (hunk 1 above)
Comment (GPUCode) [reply src/core/hle/result.h:276]: Yup copy pasta mistake

**Review by Steveice10 — APPROVED** [review]

File: src/core/file_sys/archive_ncch.cpp [hunk 3]
```diff
@@ -253,14 +251,14 @@ ResultVal<std::size_t> NCCHFile::Read(const u64 offset, const std::size_t length
     std::size_t copy_size = std::min(length, available_size);
     memcpy(buffer, file_buffer.data() + offset, copy_size);
 
-    return MakeResult<std::size_t>(copy_size);
+    return copy_size;
 }
 
 ResultVal<std::size_t> NCCHFile::Write(const u64 offset, const std::size_t length, const bool flush,
                                        const u8* buffer) {
     LOG_ERROR(Service_FS, "Attempted to write to NCCH file");
```
Comment (SachinVin) [inline src/core/file_sys/archive_ncch.cpp:259]: Unrelated code, but I guess this should say stubbed

**Pull request merged**